pkg/storageos/deploy_test.go

```diff
@@ -861,6 +861,114 @@ func TestDeployNodeAffinity(t *testing.T) {
 	}
 }
 
+func TestDeployTolerations(t *testing.T) {
+	testCases := []struct {
+		name        string
+		tolerations []corev1.Toleration
+		wantError   bool
+	}{
+		{
+			name: "TolerationOpExists without value",
+			tolerations: []corev1.Toleration{
+				{
+					Key:      "foo",
+					Operator: corev1.TolerationOpExists,
+					Effect:   corev1.TaintEffectNoSchedule,
+				},
+			},
+		},
+		{
+			name: "TolerationOpExists with value",
+			tolerations: []corev1.Toleration{
+				{
+					Key:      "foo",
+					Operator: corev1.TolerationOpExists,
+					Value:    "bar",
+					Effect:   corev1.TaintEffectNoSchedule,
+				},
+			},
+			wantError: true,
+		},
+		{
+			name: "TolerationOpEqual",
+			tolerations: []corev1.Toleration{
+				{
+					Key:      "foo",
+					Operator: corev1.TolerationOpEqual,
+					Value:    "bar",
+					Effect:   corev1.TaintEffectNoSchedule,
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			stosCluster := &api.StorageOSCluster{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: gvk.GroupVersion().String(),
+					Kind:       gvk.Kind,
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "teststos",
+					Namespace: "default",
+				},
+				Spec: api.StorageOSClusterSpec{
+					CSI: api.StorageOSClusterCSI{
+						Enable: false,
+					},
+					Tolerations: tc.tolerations,
+				},
+			}
+
+			c := fake.NewFakeClientWithScheme(testScheme)
+			if err := c.Create(context.Background(), stosCluster); err != nil {
+				t.Fatalf("failed to create storageoscluster object: %v", err)
+			}
+
+			deploy := NewDeployment(c, stosCluster, nil, testScheme, "", false)
+			err := deploy.Deploy()
+			if !tc.wantError && err != nil {
+				t.Errorf("expected no error but got one: %v", err)
+			}
+			if tc.wantError && err == nil {
+				t.Errorf("expected error but got none")
+			}
+
+			if tc.wantError {
+				return
+			}
+
+			createdDaemonset := &appsv1.DaemonSet{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "apps/v1",
+					Kind:       "DaemonSet",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      daemonsetName,
+					Namespace: stosCluster.Spec.GetResourceNS(),
+				},
+			}
+
+			nsName := types.NamespacedName{
+				Name:      daemonsetName,
+				Namespace: defaultNS,
+			}
+
+			if err := c.Get(context.Background(), nsName, createdDaemonset); err != nil {
+				t.Fatal("failed to get the created daemonset", err)
+			}
+
+			podSpec := createdDaemonset.Spec.Template.Spec
+
+			if !reflect.DeepEqual(podSpec.Tolerations, stosCluster.Spec.Tolerations) {
+				t.Errorf("unexpected Tolerations value:\n\t(GOT) %v\n\t(WNT) %v", podSpec.Tolerations, stosCluster.Spec.Tolerations)
+			}
```

Comment:
We should check the same for the statefulset too, in case of CSI only.
I'll add that in a follow-up PR.